src/main/Application/Subscriptions/SubscriptionApplicationService.cs

```diff
@@ -1,33 +1,55 @@
 ﻿using ei8.Cortex.Diary.Domain.Model;
 using ei8.Cortex.Diary.Nucleus.Client.In;
 using ei8.Cortex.Subscriptions.Common;
+using ei8.Cortex.Subscriptions.Common.Receivers;
+using Microsoft.Extensions.DependencyInjection;
+using System;
 using System.Threading.Tasks;
 
 namespace ei8.Cortex.Diary.Application.Subscriptions
 {
     public class SubscriptionApplicationService : ISubscriptionApplicationService
     {
-		private readonly ISubscriptionClient subscriptionClient;
+		private readonly IServiceProvider services;
         private readonly ITokenManager tokenManager;
 
-        public SubscriptionApplicationService(ISubscriptionClient subscriptionClient, ITokenManager tokenManager)
+        public SubscriptionApplicationService(IServiceProvider services, ITokenManager tokenManager)
 		{
-			this.subscriptionClient = subscriptionClient;
+			this.services = services;
             this.tokenManager = tokenManager;
         }
 
-		public async Task SubscribeAsync(string avatarUrl, string avatarSnapshotUrl, string deviceName, string pushAuth, string pushP256dh, string pushEndpoint)
+		public async Task SubscribeWithBrowserAsync(string avatarUrl, string avatarSnapshotUrl, string deviceName, string pushAuth, string pushP256dh, string pushEndpoint)
```

Comment:
Perhaps we can simplify this to - 
`SubscribeWithBrowserAsync(string avatarUrl, IReceiverInfo receiverInfo)`? Then depending on the type of receiverInfo invoke the appropriate subscriptionClient that was injected in the constructor.

Reply:
We can do this, but will have to revert the type-specific methods in [`ISubscriptionApplicationService.cs`](https://github.com/ei8/cortex-diary/pull/57/files#diff-2618bce9c8487bf95700cfc9db744bc2b22559b4d7a53ea09361aaa6c56d1556)

So instead we will have a single `SubscribeAsync(string avatarUrl, SubscriberInfo subscriberInfo, IReceiverInfo receiverInfo)`. By extension this will require `SubscriptionApplicationService` to be generic.

Reply:
Yes, hopefully this will relieve the App Service from the responsibility of mapping input parameters to DTOs etc (lines 33-38).

Reply:
Changed `ISubscriptionApplicationService` to be generic and added a concrete implementation for `BrowserReceiverInfo`